src/schema/submissions.ts

```diff
@@ -0,0 +1,113 @@
+import { Post, Submission, User } from './../entity';
+import { IResolvers } from 'graphql-tools';
+import { traceResolvers } from './trace';
+import { Context } from '../Context';
+import { getDiscussionLink, isValidHttpUrl } from '../common';
+import { ValidationError } from 'apollo-server-errors';
+import { GQLEmptyResponse } from './common';
+
+interface GQLArticleSubmission {
+  url: string;
+}
+
+interface GQLSubmissionAvailability {
+  hasAccess: boolean;
+  limit: number;
+  todaySubmissionsCount: number;
+}
+
+export const DEFAULT_SUBMISSION_LIMIT = '3';
+
+export const typeDefs = /* GraphQL */ `
+  type SubmissionAvailability {
+    hasAccess: Boolean!
+    limit: Int!
+    todaySubmissionsCount: Int!
+  }
+
+  extend type Query {
+    """
+    Information regarding the access of submitting community links
+    """
+    submissionsAndLimit: SubmissionAvailability!
+  }
+  extend type Mutation {
+    """
+    Submit an article to surface on users feed
+    """
+    submitArticle(url: String!): EmptyResponse @auth
+  }
+`;
+
+export const resolvers: IResolvers<unknown, Context> = traceResolvers({
+  Query: {
+    submissionsAndLimit: async (
+      _,
+      __,
+      ctx,
+    ): Promise<GQLSubmissionAvailability> => {
+      const limit = parseInt(
+        process.env.SCOUT_SUBMISSION_LIMIT || DEFAULT_SUBMISSION_LIMIT,
+      );
+
+      const repo = ctx.getRepository(Submission);
+      const user = await ctx.getRepository(User).findOne({ id: ctx.userId });
+      if (!user) {
+        return { limit, hasAccess: false, todaySubmissionsCount: 0 };
+      }
+
+      const timezone = user.timezone ?? 'utc';
+      const nowTimezone = `timezone('${timezone}', now())`;
+      const atTimezone = `at time zone '${timezone}'`;
+      const submissions = await repo.find({
+        where: `date_trunc('day', "createdAt")::timestamptz ${atTimezone} = date_trunc('day', ${nowTimezone})::timestamptz`,
+      });
+      const hasAccess = true; // subject for change on how we define the access
+
+      return { limit, todaySubmissionsCount: submissions.length, hasAccess };
+    },
+  },
+  Mutation: {
+    submitArticle: async (
```

Comment:
In case the url already exists we want to return the post itself.
So we need to return either a post or a submission object.
We can create a parent object and use a parent object something like `{result: 'succeed' | 'exists', post?: Post, submission?: Submission}` (just an example)

Reply:
We are returning the post below, in this case as the simple ValidationError as we parse the full post for now.
Is your suggestion more around a more object approach?

Reply:
Yes. I suggest that in both cases the status code will be 200 and the difference will be the object. I find it easier for the client-side to handle known issues and parse the payload

Reply:
Done, added some different flows for both the submission and the newPost routes.